src/palace/manager/api/circulation.py

```diff
@@ -1045,30 +1068,26 @@ def borrow(
                 # API does something unusual like return LoanInfo instead
                 # of raising AlreadyCheckedOut.
                 new_loan = True
+                loan_info = checkout_result
+                hold_info = None
         except AlreadyCheckedOut:
             # This is good, but we didn't get the real loan info.
             # Just fake it.
-            identifier = licensepool.identifier
-            loan_info = LoanInfo(
-                licensepool.collection,
-                licensepool.data_source,
-                identifier.type,
-                identifier.identifier,
+            loan_info = LoanInfo.from_license_pool(
+                licensepool,
                 start_date=None,
                 end_date=now + datetime.timedelta(hours=1),
+                external_identifier=(
+                    existing_loan.external_identifier if existing_loan else None
+                ),
             )
-            if existing_loan:
-                loan_info.external_identifier = existing_loan.external_identifier
         except AlreadyOnHold:
             # We're trying to check out a book that we already have on hold.
-            hold_info = HoldInfo(
-                licensepool.collection,
-                licensepool.data_source,
-                licensepool.identifier.type,
-                licensepool.identifier.identifier,
-                None,
-                None,
-                None,
+            hold_info = HoldInfo.from_license_pool(
+                licensepool,
+                start_date=None,
+                end_date=None,
+                hold_position=None,
```

Comment:
Can these lines be removed since `None` is the default for start_date, end_date and hold_position?